lib/matplotlib/image.py

```diff
@@ -396,6 +396,22 @@ def _make_image(self, A, in_bbox, out_bbox, clip_bbox, magnification=1.0,
                 # scaled data
                 A_scaled = np.empty(A.shape, dtype=scaled_dtype)
                 A_scaled[:] = A
+                # clip scaled data around norm if necessary.
+                # This is necessary for big numbers at the edge of
+                # float64's ability to represent changes.  Applying
+                # a norm first would be good, but ruins the interpolation
+                # of over numbers.
+                if self.norm.vmin is not None and self.norm.vmax is not None:
+                    dv = self.norm.vmax - self.norm.vmin
+                    vmid = self.norm.vmin + dv / 2
+                    newmin = vmid - dv * 1.e7
+                    if newmin > a_min:
+                        A_scaled[A_scaled < newmin ] = newmin
+                        a_min = np.float64(newmin)
+                    newmax = vmid + dv * 1.e7
```

Comment:
Are these newmax/newmin appropriate if `LogNorm` or something similarly non-linear is being used?

Reply:
Grrrrr, not sure.  Probably not!

Of course, the way the interpolation is done is questionable if you use LogNorm.  Right now we are doing the interpolation in data space instead of normed-space.  It seems the latter is more appropriate for non-linear norms.  ping @tacaswell

Reply:
can we use `np.clip` here?

Do we actually have to do this indexing step or can we reset `a_min` and `a_max` and let Agg deal with the clipping to [0, 1] internally?

Reply:
To be clear, I'm not saying we shouldn't take your approach--just wondering if a check is in order.

Reply:
In `LogNorm` `vmin` and `vmax` are in data space, not the log of the data, so it should work fine for `LogNorm`.  Not sure about all the other norms.

Reply:
The norms all work with `vmin` and `vmax` in data space (rather than transformed space), so this fix is fine.  I've uploaded a version that uses `np.clip` instead of indexing.  

@tacaswell  we indeed need to clip right away - not clipping is the whole problem with the floating point math of the next step.